src/test/java/com/accesspoint/rulesengine/apitests/ruleset/RulesetTest.java

```diff
@@ -466,4 +468,94 @@ public void givenRuleset_whenPOSTRulesConditionsIsEmpty_thenCustomErrorIsCalled(
                 .statusCode(400)
                 .body(equalTo("Conditions cannot be empty"));
     }
+
+    // TODO: deletes ruleset
+    @Test
+    public void givenRuleset_whenDELETE_thenCorrectDataIsPassed() {
+
+        // Building out the mock ruleset
+        Condition fakeCon1_1 = null;
+        Condition fakeCon2_1 = new Condition(200L, FRONT, EMPTY, null);
+        Condition fakeCon2_2 = new Condition(201L, LEFT, EMPTY, null);
+        List<Condition> fakeCondition1 = new ArrayList<>();
+        fakeCondition1.add(fakeCon1_1);
+        List<Condition> fakeCondition2 = new ArrayList<>();
+        fakeCondition2.add(fakeCon2_1);
+        fakeCondition2.add(fakeCon2_2);
+        Rule fakeRule1 = new Rule(100L, 11, FORWARD, null, null);
+        Rule fakeRule2 = new Rule(200L, 0, FORWARD, null, fakeCondition2);
+        List<Rule> fakeRules = new ArrayList<>();
+        fakeRules.add(fakeRule1);
+        fakeRules.add(fakeRule2);
+        Ruleset ruleset =
+                Ruleset.builder()
+                        .name("Test1")
+                        .rules(fakeRules)
+                        .id(100L)
+                        .creation_date(Timestamp.valueOf("2000-01-01 01:15:30.500"))
+                        .build();
+
+        // Mock ALL repository methods that get called in the service
+
+        when(rulesetRepository.save(Mockito.any(Ruleset.class))).thenReturn(ruleset);
```

Comment:
Not super super important to do for now, but the test should only contain what is needed for the test. So for example, this test is triggering a delete request, and the only repository methods used in a delete request would be the repository.findById() i believe. So, all of the other mocks do not need to exist.